### ISSUE TITLE
Test failure: Interop\\PInvoke\\Generics\\GenericsTest\\GenericsTest.cmd

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#36965]
failed in job: [runtime-coreclr jitstress2-jitstressregs 20200524.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=658473&view=ms.vss-test-web.build-test-results-tab&runId=20435328&resultId=100418&paneView=debug)

failed tests:
Interop\\PInvoke\\Generics\\GenericsTest\\GenericsTest.cmd
Interop\\PInvoke\\Decimal\\DecimalTest\\DecimalTest.cmd
Interop\\PInvoke\\Vector2_3_4\\Vector2_3_4\\Vector2_3_4.cmd
Interop\\PInvoke\\DateTime\\DateTimeTest\\DateTimeTest.cmd

Error message
~~~
Return code: 1
Raw output file: C:\h\w\A6980933\w\C0510A35\e\Interop\PInvoke\Reports\Interop.PInvoke\Generics\GenericsTest\GenericsTest.output.txt
Raw output:
BEGIN EXECUTION
 "C:\h\w\A6980933\p\corerun.exe" GenericsTest.dll 
Testing IComInterface<bool>
Testing IComInterface<char>
Testing IComInterface<double>
Testing IComInterface<float>
Testing IComInterface<long>
Testing IComInterface<uint>

Testing Nullable<bool>
Testing Nullable<char>
Testing Nullable<double>
Testing Nullable<float>
Testing Nullable<long>
Testing Nullable<uint>

Testing Point1<bool>
Testing Point1<char>
Testing Point1<double>
TestLibrary.AssertTestException: Assert.AreEqual: Expected: [8.243062109593103E-295]. Actual: [1]. 
 at TestLibrary.Assert.AreEqual[T](T expected, T actual, String message) in F:\workspace\_work\1\s\src\coreclr\tests\src\Common\CoreCLRTestLibrary\Assertion.cs:line 414
 at GenericsTest.TestPoint1D() in F:\workspace\_work\1\s\src\coreclr\tests\src\Interop\PInvoke\Generics\GenericsTest.Point1D.cs:line 44
 at GenericsTest.Main(String[] args) in F:\workspace\_work\1\s\src\coreclr\tests\src\Interop\PInvoke\Generics\GenericsTest.cs:line 87
Expected: 100
Actual: 0
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=C:\h\w\A6980933\p
> C:\h\w\A6980933\w\C0510A35\e\Interop\PInvoke\Generics\GenericsTest\GenericsTest.cmd
Expected: True
Actual: False


Stack trace
   at Interop_PInvoke._Generics_GenericsTest_GenericsTest_._Generics_GenericsTest_GenericsTest_cmd() in F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.x86.Checked\TestWrappers\Interop.PInvoke\Interop.PInvoke.XUnitWrapper.cs:line 6348
~~~

**Comment by AaronRobinsonMSFT** [comment]
This seems to be related to some floating point behavioral change. I don't see where that was done though.

@tannergooding Are you aware of any changes in the domain of floating point numbers?

**Comment by tannergooding** [comment]
@AaronRobinsonMSFT, the most likely thing I'm aware of is the changes around multireg struct returns: https://github.com/dotnet/runtime/issues/36912
There was also been various work in rewriting the `System.Numerics` intrinsics to be done in terms of HWIntrinsics, but it should have no impact on the `Point1<double>` failure above, for example.

CC. @sandreenko, @CarolEidt

**Comment by CarolEidt** [comment]
Fixed by #36994 or #37062
